src/coreclr/src/vm/interpreter.cpp

```diff
@@ -279,7 +279,7 @@ void InterpreterMethodInfo::InitArgInfo(CEEInfo* comp, CORINFO_METHOD_INFO* meth
                 }
                 m_argDescs[k].m_typeStackNormal = m_argDescs[k].m_type;
                 m_argDescs[k].m_nativeOffset = argOffsets_[k];
-                m_argDescs[k].m_directOffset = reinterpret_cast<short>(ArgSlotEndianessFixup(directOffset, sizeof(void*)));
+                m_argDescs[k].m_directOffset = (short) reinterpret_cast<intptr_t>(ArgSlotEndianessFixup(directOffset, sizeof(void*)));
```

Comment:
It seems very odd to have a new-style cast followed by an old-style cast. Was there a compiler error here (and below)?

Reply:
Yes, there were compiler errors (at least on Windows) concerning the original lines of type cast `reinterpret_cast<short>(...)`. The modifications fix these compiler errors.

Reply:
Would it be better, then, to have, for example:
```
m_argDescs[k].m_directOffset = reinterpret_cast<short>( reinterpret_cast<intptr_t>(ArgSlotEndianessFixup(directOffset, sizeof(void*))) );
```
to not mix the cast types?

Reply:
Hi @BruceForstall, I don't think your proposed solution compiles (on Windows). The C++ compiler returns the following error:
```
    16>C:\Users\t-yucpan\GitHub\runtime\src\coreclr\src\vm\interpreter.cpp(282,150): error C2440: 'reinterpret_cast': cannot convert from 'intptr_t' to 'short' [C:\Users\t-yucpan\GitHub\runtime\artifacts\obj\coreclr\Windows_NT.x64.Checked\src\vm\wks\cee_wks.vcxproj]
```

Reply:
maybe `static_cast` is appropriate?

Reply:
Hi @BruceForstall. Thanks for pointing this out! Indeed `static_cast<short>` works. I have converted all C-style casts to `static_cast` in commit https://github.com/dotnet/runtime/pull/37825/commits/76425bd4e09b96ba3a4dafb2a212747f22860827.